app/blueprints/assessments/activity_trail.py

```diff
@@ -0,0 +1,307 @@
+        else:
+            return []
+
+
+@dataclass
+class Comments(BaseModel):
+    id: str
+    comment: str
+    comment_type: str
+    date_created: str
+    sub_criteria_id: str
+    theme_id: str
+    user_id: str
+    full_name: str = ""
+    highest_role: str = ""
+    email_address: str = ""
+
+    @classmethod
+    def process_comments(cls, comments_list: list) -> list[dict]:
+        """
+        Retrieve all comments from updates and add the following attributes to each comment:
+        - "user_id"
+        - "sub_criteria_id"
+        - "theme_id"
+        - "comment_type"
+        - "full_name"
+        - "email_address"
+        - "highest_role"
+
+        Args: comments_list (list): A list of comments.
+
+        Returns:
+            list[dict]: A list of dictionaries containing comments with additional attributes.
+        """
+        result = []
+
+        for comment in comments_list:
+            for update in comment.get("updates"):
+                updated_update = {
+                    **update,
+                    "user_id": comment.get("user_id"),
+                    "sub_criteria_id": comment.get("sub_criteria_id"),
+                    "theme_id": comment.get("theme_id"),
+                    "comment_type": comment.get("comment_type"),
+                    "full_name": comment.get("full_name"),
+                    "email_address": comment.get("email_address"),
+                    "highest_role": comment.get("highest_role"),
+                }
+                current_app.logger.info(
+                    "Adding required information to comment"
+                )
+                result.append(updated_update)
+
+        return result
+
+    @classmethod
+    def from_list(cls, comments_list: list[dict]):
+        if comments_list:
+            comments = cls.process_comments(comments_list)
+            return super().from_list(comments)
+        else:
+            return []
+
+
+@dataclass
+class Scores(BaseModel):
+    application_id: str
+    date_created: str
+    id: str
+    justification: str
+    score: int
+    sub_criteria_id: str
+    user_id: str
+    full_name: str = ""
+    highest_role: str = ""
+    email_address: str = ""
+
+
+def get_user_info(list_data: list, state: AssessorTaskList) -> dict:
+
+    if list_data is None:
+        return []
+
+    user_list = []
+    for item in list_data:
+        user_id = item.user_id
+        if user_id is not None and user_id not in user_list:
+            user_list.append(user_id)
+
+    if user_list:
+        current_app.logger.info("Retrieving account information")
+        account_list = get_bulk_accounts_dict(
+            user_list,
+            state.fund_short_name,
+        )
+        return account_list
+    else:
+        current_app.logger.warning(
+            "Could not retrieve the account information"
+        )
+        return []
+
+
+def add_user_info(list_data: list, state, account_info: list = None) -> list:
```

Comment:
`account_info` looks to be immediately re-assigned - do we need it here?

Reply:
yea but in the function, it has `account_info: list = None` but then the line after `account_info = get_user_info(list_data, state)`, so the first part is unnecessary, cause it's immediately re-assigned

Reply:
Aah, I see what you mean 😆 .  Sorry missed that part 🙏